recipes/fast-dds/all/conanfile.py

```diff
@@ -144,10 +140,20 @@ def validate(self):
             if ("MT" in self.settings.compiler.runtime and self.options.shared):
                 # This combination leads to an fast-dds error when linking
                 # linking dynamic '*.dll' and static MT runtime
-                raise ConanInvalidConfiguration("Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
+                raise ConanInvalidConfiguration(
+                    "Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
 
     def build(self):
-        self._patch_sources()
+        # dynamic patch works for 2.3.2 to 2.5.0
+        tools.replace_in_file("{}/CMakeLists.txt".format(self._source_subfolder),
+                              "eprosima_find_thirdparty(TinyXML2 tinyxml2)",
+                              "eprosima_find_thirdparty(tinyxml2 REQUIRED)")
+        tools.replace_in_file("{}/CMakeLists.txt".format(self._source_subfolder),
+                              "eprosima_find_thirdparty(Asio asio VERSION 1.10.8)",
+                              "eprosima_find_thirdparty(asio REQUIRED)")
```

Comment:
There's currently an error in the Fast-DDS patches.
See the code for `eprosima_find_thirdparty` here: https://github.com/eProsima/Fast-DDS/blob/master/cmake/common/eprosima_libraries.cmake#L168

The second argument shouldn't be `REQUIRED` but rather the name of the library.
It's not really important considering how we are using it here, but the patches should be changed into
```
eprosima_find_thirdparty(tinyxml2 tinyxml2)
eprosima_find_thirdparty(asio asio)
```